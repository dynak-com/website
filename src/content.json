[
  {
    "lang": "pl",
    "nav": [
      {
        "text": "nav.home",
        "address": "/"
      },
      {
        "text": "nav.about",
        "address": "/about"
      },
      {
        "text": "nav.projects",
        "address": "/projects"
      },
      {
        "text": "nav.contact",
        "address": "/contact"
      }
    ],
    "home": {
      "header": "home.header",
      "posts": [
        {
          "id": 1,
          "text": "home.posts.paragraph1",
          "headerText": "home.posts.headerText1"
        },
        {
          "id": 2,
          "text": "home.posts.paragraph2",
          "headerText": "home.posts.headerText2"
        },
        {
          "id": 3,
          "text": "home.posts.paragraph3",
          "headerText": "home.posts.headerText3"
        }
      ]
    },
    "about": {
      "header": "about.header",
      "technologies": {
        "title": "about.technologies.title",
        "content": "about.technologies.content"
      },
      "team": {
        "title": "about.team.title",
        "teamData": [
          {
            "name": "Emilia Dynak",
            "picture": "https://secure.gravatar.com/avatar/d34a1b21822840695b16113034ba740b?s=180&d=identicon",
            "position": "CEO and co-owner",
            "linkedInUrl": "https://www.linkedin.com/in/emilia-dynak-b0a3a9140/",
            "githubUrl": "https://github.com/edynak",
            "username": "edynak"
          },
          {
            "name": "Cezary Dynak",
            "picture": "https://secure.gravatar.com/avatar/47fbd06257cedcf9594bd9144f948feb?s=180&d=identicon",
            "position": "CTO and co-owner",
            "linkedInUrl": "https://www.linkedin.com/in/cdynak/",
            "githubUrl": "https://github.com/cdynak",
            "username": "cdynak"
          },
          {
            "name": "Natalia Borzyszkowska",
            "picture": "https://secure.gravatar.com/avatar/2cba91020f68596ad56766847694b292?s=180&d=identicon",
            "position": "CFO",
            "linkedInUrl": "https://www.linkedin.com/in/natalia-borzyszkowska-b7814218b/",
            "githubUrl": "https://github.com/login",
            "username": "21_smakow_natki"
          },
          {
            "name": "Aleksandra Borodziuk",
            "picture": "https://secure.gravatar.com/avatar/865d01afdfe703b4ef137693198b8500?s=180&d=identicon",
            "position": "CMO",
            "linkedInUrl": "https://www.linkedin.com/in/aleksandra-borodziuk/",
            "githubUrl": "https://github.com/login",
            "username": "Forestgirl25"
          },
          {
            "name": "Dominika Konefał",
            "picture": "https://secure.gravatar.com/avatar/ec094fb729d8e3509391628d8ba00dd8?s=180&d=identicon",
            "position": "Graphic Designer",
            "linkedInUrl": "https://www.linkedin.com/in/dominika-konefa%C5%82-028920185/",
            "githubUrl": "https://github.com/login",
            "username": "Deloi"
          },
          {
            "name": "Mikołaj Tumalewicz",
            "picture": "https://gitlab.com/uploads/-/system/user/avatar/2270247/avatar.png?width=180",
            "position": "DevOps/SysOps Engineer",
            "linkedInUrl": "https://www.linkedin.com/in/mtumalewicz/",
            "githubUrl": "https://github.com/mtumalewicz",
            "username": "mtumalewicz"
          },
          {
            "name": "Marcin Banik",
            "picture": "https://gitlab.com/uploads/-/system/user/avatar/2342806/avatar.png?s=180",
            "position": "Front-end Developer",
            "linkedInUrl": "https://www.linkedin.com/in/marcin-banik-947954142/",
            "githubUrl": "https://github.com/m-banik",
            "username": "mbanik"
          },
          {
            "name": "Bartosz Marciniec",
            "picture": "https://gitlab.com/uploads/-/system/user/avatar/4684288/avatar.png?s=180",
            "position": "Back-end Developer",
            "linkedInUrl": "https://www.linkedin.com/in/bartosz-marciniec-b0917517a/",
            "githubUrl": "https://github.com/hendas",
            "username": "hendas"
          },
          {
            "name": "Cristina Volk",
            "picture": "https://assets.gitlab-static.net/uploads/-/system/user/avatar/5111565/avatar.png?s=180",
            "position": "Web Developer",
            "linkedInUrl": "https://www.linkedin.com/in/kristina-volk-a9829517b/",
            "githubUrl": "https://github.com/CristinaVolk",
            "username": "CristinaVolk"
          }
        ]
      },
      "values": {
        "title": "about.values.title",
        "content": [
          "about.values.content.paragraph1",
          "about.values.content.paragraph2",
          "about.values.content.paragraph3"
        ]
      }
    },
    "projects": {
      "header": "projects.header",
      "posts": [
        {
          "id": 1,
          "text": "projects.posts.text1",
          "headerText": "dynak.io"
        },
        {
          "id": 2,
          "text": "projects.posts.text2",
          "headerText": "warehouse-app"
        },
        {
          "id": 3,
          "text": "projects.posts.text3",
          "headerText": "projects.posts.headerText3"
        }
      ]
    },
    "contact": {
      "header": "contact.header",
      "socials": "contact.socials",
      "form": {
        "subject": "contact.form.subject",
        "email": "contact.form.email",
        "message": "contact.form.message",
        "contactAgreement": "contact.form.contactAgreement",
        "privacyPolicyAgreement": "contact.form.privacyPolicyAgreement",
        "policyPrivacy": "contact.form.policyPrivacy",
        "send": "contact.form.send"
      }
    },
    "modal": {
      "modal_ok": {
        "modal_id": "modal_ok",
        "error_message": false,
        "thanks": "modal_ok.thanks",
        "sent": "modal_ok.sent",
        "close": "modal_ok.close"
      },
      "modal_error": {
        "modal_id": "modal_error",
        "error_message": true,
        "thanks": "modal_error.thanks",
        "sent": "modal_error.sent",
        "close": "modal_error.close"
      },
      "modal_privacy_policy": {
        "modal_id": "modal_privacy_policy",
        "error_message": null,
        "thanks": "modal_privacy_policy.thanks",
        "sent": "modal_privacy_policy.sent",
        "close": "modal_privacy_policy.close",
        "rodoContent": {
          "header": "rodoContent.header",
          "sentence1": "rodoContent.sentence1",
          "website": "rodoContent.website",
          "sentence2": "rodoContent.sentence2",
          "generalProvisions": "rodoContent.generalProvisions",
          "text1": "rodoContent.text1",
          "text2": "rodoContent.text2",
          "text3": "rodoContent.text3",
          "definitions": "rodoContent.definitions",
          "administrator": "rodoContent.administrator",
          "user": "rodoContent.user",
          "personalData": "rodoContent.personalData",
          "paragraph1": "rodoContent.paragraph1",
          "linkedIn": "rodoContent.linkedIn",
          "paragraph2": "rodoContent.paragraph2",
          "paragraph3": "rodoContent.paragraph3",
          "paragraph4": "rodoContent.paragraph4",
          "paragraph5": "rodoContent.paragraph5",
          "paragraph6": "rodoContent.paragraph6",
          "paragraph7": "rodoContent.paragraph7",
          "paragraph8": "rodoContent.paragraph8",
          "webLinkedIn": "rodoContent.webLinkedIn",
          "github": "rodoContent.github",
          "gitlab": "rodoContent.gitlab",
          "paragraph9": "rodoContent.paragraph9",
          "paragraph10": "rodoContent.paragraph10",
          "paragraph11": "rodoContent.paragraph11",
          "paragraph12": "rodoContent.paragraph12",
          "paragraph13": "rodoContent.paragraph13",
          "paragraph14": "rodoContent.paragraph14",
          "paragraph15": "rodoContent.paragraph15",
          "paragraph16": "rodoContent.paragraph16",
          "paragraph17": "rodoContent.paragraph17",
          "paragraph18": "rodoContent.paragraph18",
          "paragraph19": "rodoContent.paragraph19",
          "paragraph20": "rodoContent.paragraph20",
          "paragraph21": "rodoContent.paragraph21",
          "paragraph22": "rodoContent.paragraph22",
          "paragraph23": "rodoContent.paragraph23",
          "forms": "rodoContent.forms",
          "paragraph24": "rodoContent.paragraph24",
          "paragraph25": "rodoContent.paragraph25",
          "technologies": "rodoContent.technologies",
          "Administrator": "rodoContent.Administrator",
          "not": "rodoContent.not",
          "paragraph26": "rodoContent.paragraph26",
          "cookies": "rodoContent.cookies",
          "part1": "rodoContent.part1",
          "part2": "rodoContent.part2",
          "part3": "rodoContent.part3",
          "part4": "rodoContent.part4",
          "created": "rodoContent.created"
        }
      }
    },
    "error": "error",
    "footer": {
      "rodo": "footer.rodo",
      "dynak": "footer.dynak"
    }
  }
]