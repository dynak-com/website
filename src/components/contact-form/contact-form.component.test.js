import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import ContactForm from './contact-form.component';
import content from '../../content.json';
import useComponent from './hook';

const [contentData] = content;
const { form } = contentData.contact;

function HookWrapper(props) {
    const hook = props.hook ? props.hook() : undefined;
    return <div hook={hook} />;
}

describe('ContactForm  component', () => {
    it('renders correctly according to Snapshot', () => {
        const tree = renderer.create(<ContactForm content={form} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('ContactForm ', () => {
    it('should render', () => {
        const wrapper = shallow(<HookWrapper />);

        expect(wrapper.exists()).toBeTruthy();
    });

    it('should set init value', () => {
        const wrapper = shallow(<HookWrapper hook={() => useComponent()} />);

        const { hook } = wrapper.find('div').props();
        const { state, onChangeInputDone, onSubmitFormDone, onPrivacyPolicyModalOpenDone } = hook;
        expect(state).toEqual({
            email: '',
            isContactAgreed: false,
            isPrivacyPolicyAgreed: false,
            message: '',
            subject: '',
        });
        expect(onChangeInputDone).toMatchSnapshot();
        expect(onSubmitFormDone).toMatchSnapshot();
        expect(onPrivacyPolicyModalOpenDone).toMatchSnapshot();
    });

    describe('<ContactForm callbacks />', () => {
        let wrapper;
        const setState = jest.fn();
        const useStateSpy = jest.spyOn(React, 'useState');
        useStateSpy.mockImplementation((init) => [init, setState]);

        beforeEach(() => {
            wrapper = shallow(<ContactForm content={form} />);
        });

        afterEach(() => {
            jest.clearAllMocks();
        });

        it('calls onChangeInputDone', () => {
            const mockedEvent = { target: {} };
            let i = 0;
            while (i < 6) {
                wrapper.find('input').at(i).simulate('change', mockedEvent);
                i++;
            }
            expect(setState).toHaveBeenCalledTimes(4);
        });
    });
});

it('sendData from contact form calls fetch', async () => {
    const mockSuccessResponse = {};
    const mockJsonPromise = Promise.resolve(mockSuccessResponse);
    const onError = jest.fn();

    const mockFetchPromise = Promise.resolve({
        json: () => mockJsonPromise,
    });
    jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);

    let wrapper = shallow(<ContactForm content={form} />);
    wrapper.find('form').simulate('submit', { preventDefault() {} });

    await expect(global.fetch).toHaveBeenCalledTimes(1);
    expect(onError).not.toHaveBeenCalled();
});
