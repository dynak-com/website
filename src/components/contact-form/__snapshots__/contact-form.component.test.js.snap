// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm  component renders correctly according to Snapshot 1`] = `
<div
  className="mx-auto"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-row"
    >
      <div
        className="form-group col-md-6"
      >
        <input
          className="form-control text-center"
          name="subject"
          onChange={[Function]}
          placeholder="Temat wiadomości"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group col-md-6"
      >
        <input
          className="form-control text-center"
          name="email"
          onChange={[Function]}
          placeholder="Twój adres e-mail"
          required={true}
          type="email"
          value=""
        />
      </div>
    </div>
    <textarea
      className="form-control"
      cols="30"
      name="message"
      onChange={[Function]}
      placeholder="Treść wiadomości"
      required={true}
      rows="10"
      value=""
    />
    <div
      className="form-row my-3"
    >
      <div
        className="form-group col"
      >
        <div
          className="custom-control custom-checkbox mr-sm-2"
        >
          <input
            checked={false}
            className="custom-control-input"
            id="isContactAgreed-input"
            name="isContactAgreed"
            onChange={[Function]}
            required={true}
            type="checkbox"
          />
          <label
            className="custom-control-label text-justify label"
            htmlFor="isContactAgreed-input"
          >
            Wyrażam zgodę na kontakt ze strony administratora danych osobowych, z użyciem telekomunikacyjnych urządzeń końcowych oraz automatycznych systemów wywołujących, zgodnie z art. 172 par. 1. Prawa telekomunikacyjnego.
          </label>
        </div>
        <div
          className="custom-control custom-checkbox mr-sm-2"
        >
          <input
            checked={false}
            className="custom-control-input"
            id="isPrivacyPolicyAgreed-input"
            name="isPrivacyPolicyAgreed"
            onChange={[Function]}
            required={true}
            type="checkbox"
          />
          <label
            className="custom-control-label text-justify label"
            htmlFor="isPrivacyPolicyAgreed-input"
          >
            Zapoznałem/-am się z polityką prywatności i akceptuję jej treść.
          </label>
        </div>
      </div>
      <div
        className="form-group col-md-12 text-center text-md-right buttonsPanel"
      >
        <input
          className="btn d-inline-block mx-1"
          onClick={[Function]}
          type="button"
          value="Polityka prywatności"
        />
        <input
          className="btn d-inline-block mx-1"
          id="buttonSubmit"
          type="submit"
          value="Wyślij"
        />
      </div>
    </div>
  </form>
</div>
`;

exports[`ContactForm  should set init value 1`] = `[Function]`;

exports[`ContactForm  should set init value 2`] = `[Function]`;

exports[`ContactForm  should set init value 3`] = `[Function]`;
